src/Proto3/Suite/Class.hs

```diff
@@ -405,16 +407,16 @@ instance Primitive BL.ByteString where
   primType _ = Bytes
 
 instance forall e. (Bounded e, Named e, Enum e) => Primitive (Enumerated e) where
-  encodePrimitive num = Encode.enum num . enumify . enumerated
+  encodePrimitive num = Encode.enum num . enumify . coerce @(Enumerated e) @(Either Int e)
     where enumify (Left i) = i
           enumify (Right x) = fromEnum x
-  decodePrimitive = fmap Enumerated Decode.enum
-  primType _ = Named (Single (nameOf (Proxy @e)))
+  decodePrimitive = coerce @(_ (Either Int e)) @(_ (Enumerated e)) Decode.enum
```

Comment:
I think partial type signatures in code make it more confusing to read; I can see why this is more concise when you have type-context in your head but without that context I have to do more work to figure out what those holes are. This follows the philosophy that types are documentation and explicit documentation will make the code easier to read (whether you're trying to maintain it or simply understand it).

For example, I have no idea what that type hole is and I'd be nervous to change this code without knowing it, so I stuck something I knew would give a type-error there just so I could see what the inferred type was (`Parser RawParser`). As an aside, I guess GHC doesn't provide type-hole information in type applications?

Reply:
This comment applies to all of the partial type signature instances in the code.

Reply:
About the point regarding conciseness: I understand that it feels fmappy by omitting the `Parser RawParser` but I think the partial signature produces more questions for the reader than the other way around. In fact I think the coerce is _nicer_ than the fmap version (setting aside convenience and possible efficiency reasons) from a type-based documentation perspective, if you fill in the partial type sigs.

Reply:
I originally wrote the whole type out (see commit f815255), but I felt that it was too verbose and obscured the parts of the type that was actually different.

Reply:
Yeah, as someone who may have to maintain this code, I prefer the verbose version you originally wrote.

Reply:
ok